src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -0,0 +1,960 @@
+  auto& dT2 = get<::Tags::Tempab<36, SpatialDim, Frame>>(buffer);
+  auto& dT3 = get<::Tags::Tempab<37, SpatialDim, Frame>>(buffer);
+
+  // 3+1 quantities
+  spacetime_unit_normal =
+      gr::spacetime_normal_vector<SpatialDim, Frame, DataVector>(lapse, shift);
+  spatial_metric =
+      gr::spatial_metric<SpatialDim, Frame, DataVector>(spacetime_metric);
+  inverse_spacetime_metric =
+      gr::inverse_spacetime_metric<SpatialDim, Frame, DataVector>(
+          lapse, shift, inverse_spatial_metric);
+  time_deriv_of_spatial_metric<SpatialDim, Frame, DataVector>(
+      make_not_null(&d0_spatial_metric), lapse, shift, phi, pi);
+  deriv_spatial_metric<SpatialDim, Frame, DataVector>(
+      make_not_null(&d3_spatial_metric), phi);
+
+  // commonly used terms
+  constexpr auto exp_fac_1 = 0.5;
+  constexpr auto exp_fac_2 = 0.;
+  get(one_over_lapse) = 1. / get(lapse);
+  DampedHarmonicGauge_detail::log_factor<DataVector>(
+      make_not_null(&log_fac_1), lapse, sqrt_det_spatial_metric, exp_fac_1);
+  DampedHarmonicGauge_detail::log_factor<DataVector>(
+      make_not_null(&log_fac_2), lapse, sqrt_det_spatial_metric, exp_fac_2);
+
+  // Tempering functions
+  const auto roll_on_h_init = DampedHarmonicGauge_detail::roll_on_function(
+      time, t_start_h_init, sigma_t_h_init);
+  const auto roll_on_L1 = DampedHarmonicGauge_detail::roll_on_function(
+      time, t_start_L1, sigma_t_L1);
+  const auto roll_on_L2 = DampedHarmonicGauge_detail::roll_on_function(
+      time, t_start_L2, sigma_t_L2);
+  const auto roll_on_S =
+      DampedHarmonicGauge_detail::roll_on_function(time, t_start_S, sigma_t_S);
+  DampedHarmonicGauge_detail::weight_function<SpatialDim, Frame, DataVector>(
+      make_not_null(&weight), coords, sigma_r);
+
+  // coeffs that enter gauge source function
+  get(mu_L1) =
+      amp_coef_L1 * roll_on_L1 * get(weight) * pow(get(log_fac_1), exp_L1);
+  get(mu_S) = amp_coef_S * roll_on_S * get(weight) * pow(get(log_fac_1), exp_S);
+  get(mu_L2) =
+      amp_coef_L2 * roll_on_L2 * get(weight) * pow(get(log_fac_2), exp_L2);
+  get(mu_S_over_lapse) = get(mu_S) * get(one_over_lapse);
+
+  // Calc \f$ \mu_1 = \mu_{L1} log(rootg/N) = R W log(rootg/N)^5\f$
+  get(mu1) = get(mu_L1) * get(log_fac_1);
+
+  // Calc \f$ \mu_2 = \mu_{L2} log(1/N) = R W log(1/N)^5\f$
+  get(mu2) = get(mu_L2) * get(log_fac_2);
+
+  const auto d0_roll_on_h_init =
+      DampedHarmonicGauge_detail::time_deriv_of_roll_on_function(
+          time, t_start_h_init, sigma_t_h_init);
+  const auto d0_roll_on_L1 =
+      DampedHarmonicGauge_detail::time_deriv_of_roll_on_function(
+          time, t_start_L1, sigma_t_L1);
+  const auto d0_roll_on_S =
+      DampedHarmonicGauge_detail::time_deriv_of_roll_on_function(
+          time, t_start_S, sigma_t_S);
+  const auto d0_roll_on_L2 =
+      DampedHarmonicGauge_detail::time_deriv_of_roll_on_function(
+          time, t_start_L2, sigma_t_L2);
+
+  // Calc \f$ \partial_a [R W] \f$
```

Comment:
:+1: I appreciate these code comment for the intermediate steps of the computation. It helps a lot to follow these more complicated calculations.